lib/rules/jsx-max-props-per-line.js

```diff
@@ -43,7 +57,19 @@ module.exports = {
   create(context) {
     const configuration = context.options[0] || {};
     const maximum = configuration.maximum || 1;
-    const when = configuration.when || 'always';
+    let maximumSingle = null;
+    let maximumMulti = null;
+
+    const isExtendedConfig = typeof maximum !== 'number';
+
+    if (isExtendedConfig) {
+      maximumSingle = maximum.single || 1;
+      maximumMulti = maximum.multi || 1;
+    }
+
+    const when = isExtendedConfig
+      ? 'always'
+      : configuration.when || 'always';
```

Comment:
nothing in the schema seems to prevent passing a single/multi object and also setting `when`. if it's going to be ignored then we should make that schema disallowed.